src/lib/operators/join_sort_merge.cpp

```diff
@@ -160,15 +166,33 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractJoinOperatorImpl {
 
   /**
   * Determines the number of clusters to be used for the join.
-  * The number of clusters must be a power of two, i.e. 1, 2, 4, 8, 16...
-  * TODO(anyone): How should we determine the number of clusters?
+  * This task is not trivial as multiple aspects have to be considered: (i) the system's cache
+  * size, (ii) potential partitioning overhead, and (iii) the impact on successive operators.
+  * As of now, the cache can only be estimated. A size of 256k is used as this should be close 
+  * to the working machine of the students. For servers, however, this number might be vastly off.
+  * Aspects (i) and (ii) determine the performance of the join alone. Many partitions
```

Comment:
How much of a difference does correctly "choosing" the cache size make? Adding a helper somewhere that tells us exactly that should not be more than a hundred lines.

Reply:
It's dozens of percent for the sole join runtime. For the hash join, I did some testing last week and when we chose the "correct" size for nemea, it brings more then 10% for Q9.

Reply:
Do you want to open an issue for this (or do it right away)? We should introduce some `HardwareIdentifier` (name tbd) that exposes not only this, but also the number of cores as used by the scheduler and whatever else we might need.

Reply:
I'll dig up the number for the hash join again and open an issue.